math/mathcore/inc/Math/FitMethodFunction.h

```diff
@@ -64,10 +66,42 @@ class BasicFitMethodFunction : public FunctionType {
       method returning the data i-th contribution to the fit objective function
       For example the residual for the least square functions or the pdf element for the
       likelihood functions.
-      Estimating eventually also the gradient of the data element if the passed pointer  is not null
+      Estimating also the gradient of the data element if the passed pointer  is not null
+      and the Hessian. The flag fullHessian is set when one needs to compute the full Hessian (not the approximated one)
+      and should be used when the full second derivatives of the model functions are available
     */
-   virtual double DataElement(const double *x, unsigned int i, double *g = nullptr) const = 0;
+   virtual double DataElement(const double *x, unsigned int i, double *g = nullptr, double *h = nullptr, bool fullHessian = false) const = 0;
```

Comment:
I don't understand completely what the interface expects you to write into `h` now.

I guess it is the full 2D Hession when `fullHessian` is true, otherwise it's just the diagonal element? It would be good for the case where one wants to overload this function to understand better what the output parameters expect, and the documentation only mentions some approximated Hessian where I'm not sure what it is

Reply:
I agree it should document this better. 
The fullHessian flag indicates to compute the Hessian of the Likelihood Element using the Hessian of the model function, in the approximated case we use only the first derivatives of the model function (see page 31 of https://seal.web.cern.ch/documents/minuit/mntutorial.pdf) and this approximation is used in Minuit2/Fumili, in TFumili and in GSLNLSMinimizer.